File: c/chapitre_3/exo_32.c
/*
Ecrire un programme qui prend en argument un chemin vers un répertoire R, et copie le répertoire courant dans ce répertoire R.
*/

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main(int argc , char **argv)
{
    if (argc != 2 )
    {
        perror("Inserez le chemin vers le repertoire\n");
        exit(-1);
    }
    else
    {
        execl("/usr/bin/cp" , "cp" , "-r" , getenv("PWD")  , argv [1] , NULL ) ;
    }
    
    return 0 ;
}

File: python/chapitre_3/exo32.py
"""
Ecrire un programme qui prend en argument un chemin vers un répertoire R, et copie le répertoire courant dans ce répertoire R.
"""

import os
import sys
import shutil

def main():
    if len(sys.argv) != 2:
        print("Inserez le chemin vers le repertoire")
        sys.exit(-1)
    else:
        shutil.copytree(os.getcwd(), sys.argv[1], dirs_exist_ok=True)

if __name__ == "__main__":
    main()